pkg/fuzz/dataformat/multipart.go

```diff
@@ -41,11 +49,40 @@ func (m *MultiPartForm) Encode(data KV) (string, error) {
 	data.Iterate(func(key string, value any) bool {
 		var fw io.Writer
 		var err error
+
+		if filesArray, ok := value.([]interface{}); ok {
+			fileMetadata, ok := m.filesMetadata[key]
+			if !ok {
+				Itererr = fmt.Errorf("file metadata not found for key %s", key)
+				return false
+			}
+
+			for _, file := range filesArray {
+				h := make(textproto.MIMEHeader)
+				h.Set("Content-Disposition",
+					fmt.Sprintf(`form-data; name="%s"; filename="%s"`,
+						escapeQuotes(key), escapeQuotes(fileMetadata.Filename)))
```

Comment:
I think we could use [`mime.FormatMediaType`](https://pkg.go.dev/mime#FormatMediaType) here to get rid of [`quoteEscaper`](https://github.com/projectdiscovery/nuclei/pull/5702/files#diff-489fead1c8ef21824e4c2544a0840d201ff22d9ed52554c9e06c8bb6b9d44f81R100) and make it less error-prone.

Reply:
It actually doesn't do the same thing. I looked around and according to https://stackoverflow.com/a/21133168 there isn't a way to do this currently easily in go without writing `w.CreatePart(h)` as we did so we have to create the header ourselves it seems like